integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/NewCarbonDataLoadRDD.scala

```diff
@@ -386,10 +389,15 @@ class NewRddIterator(rddIter: Iterator[Row],
     val len = columns.length
     var i = 0
     while (i < len) {
-      columns(i) = CarbonScalaUtil.getString(row, i, carbonLoadModel, serializationNullFormat,
-        complexDelimiters, timeStampFormat, dateFormat,
-        isVarcharType = i < isVarcharTypeMapping.size && isVarcharTypeMapping(i),
-        isComplexType = i < isComplexTypeMapping.size && isComplexTypeMapping(i))
+      columns(i) = row.get(i) match {
+        case bs if bs.isInstanceOf[Array[Byte]] && isDefaultBinaryDecoder =>
+          bs.asInstanceOf[Array[Byte]]
```

Comment:
If this is byte[] you don't want to apply the binary decoder right ?
@niuge01  check this.

changes looks ok to me.

Reply:
Yes, in default binary decoder, `CarbonScalaUtil.getString` may destroy the byte[].

Reply:
Ideally, It should not destroy. I don't know why it will be problem.
But if the byte[] already there. no need to convert to string and get byte[] again. so merging this.